.github/workflows/wasm_editor_and_interpreter.yaml

```diff
@@ -40,24 +40,30 @@ jobs:
     - name: Install NPM dependencies
       run: npm install
       working-directory: tools/online_editor
+    - name: Build slint-wasm-interpreter
+      run: npm run build:wasm_preview-release
+      working-directory: tools/online_editor
+    - name: Build slint-wasm-lsp
+      run: npm run build:wasm_lsp-release
+      working-directory: tools/online_editor
+    - name: Lint online editor # This needs the slint-wasm-interpreter!
+      run: |
+          npm run syntax_check
+          npm run lint
+      working-directory: tools/online_editor
+    - name: Set up symlinks
+      run: |
+          ln -s slint/api/wasm-interpreter/pkg wasm-interpreter
+          ln -s slint/tools/lsp/pkg wasm-lsp
+      working-directory: ..
```

Comment:
why do one need these symlinks while they were not required before?

Reply:
I have the suspicion that the `external` magic is failing in my vite.

Reply:
If I set up external to `"../../../../wasm-interpreter/slint_wasm_interpreter.js",
`, then I do not need the `wasm-interpreter` symlink.

Unfortunately I have not managed to figure out how to do the same for `wasm-lsp`, that is not found without symlink -- even if I set the file it does not find as `external`.

Reply:
The old setting in the vite config is this:

```json
external: ["../../../wasm-interpreter/slint_wasm_interpreter.js"],
```

Would it perhaps be possible to simply add a second entry to this array for the lsp js entry point, so that rollup doesn't follow it and bundle it?

Reply:
I added the lsp as external in all the different ways I can think off, but it still insists on trying to add that:-/

This is the error message:

```
➜ npx vite build
vite v2.9.15 building for production...
✓ 47 modules transformed.
✓ 50 modules transformed.
✓ 6 modules transformed.
[vite:worker-import-meta-url] Could not load ../../../../../wasm-lsp/slint_lsp_wasm.js (imported by src/worker/lsp_worker.ts): ENOENT: no such file or directory, open '../../../../../wasm-lsp/slint_lsp_wasm.js'
file: /home/dev/src/slint/tools/online_editor/src/index.ts
error during build:
Error: Could not load ../../../../../wasm-lsp/slint_lsp_wasm.js (imported by src/worker/lsp_worker.ts): ENOENT: no such file or directory, open '../../../../../wasm-lsp/slint_lsp_wasm.js'
```

The relevant section in vite.config.js looks like this:

```js
    base_config.build = {};
    base_config.build.rollupOptions = {
      external: [
        "@preview/",
        "@lsp/",
        "slint_lsp_wasm.js",
        "wasm-lsp/slint_lsp_wasm.js",
        "../wasm-lsp/slint_lsp_wasm.js",
        "../../wasm-lsp/slint_lsp_wasm.js",
        "../../../wasm-lsp/slint_lsp_wasm.js",
        "../../../../wasm-lsp/slint_lsp_wasm.js",
        "../../../../../wasm-lsp/slint_lsp_wasm.js",
        "../../../../../../wasm-lsp/slint_lsp_wasm.js",
        "../../../../../../../wasm-lsp/slint_lsp_wasm.js",
        "../../../../../../../../wasm-lsp/slint_lsp_wasm.js",
        "../../../../../../../../../wasm-lsp/slint_lsp_wasm.js",
        "../../../../wasm-interpreter/slint_wasm_interpreter.js",
      ],
      input: ["index.html", "preview.html"],
    };
    base_config.resolve = {
      alias: {
        "@preview/": "../../../../wasm-interpreter/",
        "@lsp/": "../../../../../wasm-lsp/",
      },
    };
```

It is shrongly advised not to use relative paths since vite might get confused with those, so I added some extra ones;-)

What I did notice is that the `wasm-interpreter` produces a different error message when that is not listed as `external`:

```
[vite:load-fallback] Could not load ../../../../wasm-interpreter/slint_wasm_interpreter.js (imported by src/preview.ts): ENOENT: no such file or directory, open '../../../../wasm-interpreter/slint_wasm_interpreter.js'
error during build:
Error: Could not load ../../../../wasm-interpreter/slint_wasm_interpreter.js (imported by src/preview.ts): ENOENT: no such file or directory, open '../../../../wasm-interpreter/slint_wasm_interpreter.js'
```

So I think this might be due to the webworker magic that happens in vite (wasm-interpreter is imported into `index.ts`, while the lsp is imported in `lsp_worker.ts` which is a webworker.

Reply:
It is indeed due to the webworker: WebWorkers can not be external in vite at this time! Looking for a workaround now.

Reply:
There is apparently little we can do about this at this time short of replacing vite. I am happy to just ignore the problem for the time being: The LSP is used in this one place only anyway. So for now we do not loose too much by having it rolled into this.